### DISCUSSION TITLE
Report on my build

### DISCUSSION THREAD

**Discussion by MikeMcPherson** [alanbjohnston/CubeSatSim#150] [Show and tell]
Hi everyone

The first of two CubesatSims is finally operational!  Between university ordering snafus, parts shortages, counterfeit parts, and various family emergencies, it took far longer than it should have.  I'm really pleased with the outcome, though.  

I'm working on writing up lab exercises; we expect to be able to use the CubesatSim for the 2022/23 Spacecraft Design course at UVA, the two-semester capstone for the Aerospace Engineering BS program.  I volunteer as an advisor for communications topics.  

UVA students designed, built, and launched their first 1U in March 2019.  We managed to establish communications with the spacecraft, but the mission was truncated due to a couple of critical hardware failures.  The spacecraft re-entered and burned up early this year.

We are going to use the CubesatSims to give the students some hands-on operational experience in between missions.  (We're still looking for funding for a couple of possible missions.)

Enough background.  I was very impressed with the build documentation.  The text was very clear and the high-resolution photos were a great help.  Your build documentation is some of the best I've seen.  Thanks!

I did run into a few things I want to mention.  (I have the V1.1 boards.  See pictures below.)  First, because the purple INA breakout boards were junk, I went with the blue boards.  The four on the Pi side of the Main Board fit with no problem, but the four on the other side interfered with the JST connectors.  I ended up mounting them on pin headers, which worked well.  I had to notch two of the INA boards using a Dremel tool to fit around the spacers when I built the board stack.

I decided to mount the two sensor breakouts on the STEM board on pin headers, as well, just in case they failed.  I went with the Black Pill, and had to notch the top of the frame to clear the "user" button on the board.

I discovered that the spacer combinations specified for bottom of the stack (the two on the Pi and the other two) were too short.  I substituted 15mm in place of 10mm, and it worked out almost perfectly.  I had to trim the two not on the Pi to get things level.

I had a lot of trouble with the nylon spacers breaking, so I may try brass on the second build.

I think that's all I ran into.

A thought for the future...  Although I can see some attraction in the "kit-bashing" approach, it does increase complexity and introduces sourcing and reliability problems.  I wonder if a surface mount version would be worthwhile.  For universities, access to reflow ovens and students who know how to use them is no problem.  (Hobbyists, too.  I have a toaster oven conversion reflow oven at home.)  If you specify larger surface mount parts, then hand soldering is possible, especially for young people with steady hands.  Or, making arrangements with one of the maker-focused companies that will fabricate the board and populate it with the components could work.  I'm sure you've considered this, but I wanted to mention it.

Thanks, again, for the great design and build instructions.  I'm really looking forward to using this during the coming year.

Mike KQ9P

![IMG_1487](https://user-images.githubusercontent.com/1490116/176199006-3222b70c-75a1-4ce0-bbe1-a334073e3005.JPG)
![IMG_1488](https://user-images.githubusercontent.com/1490116/176199016-3c812b32-5857-41ce-a70d-8e793e9123a6.JPG)
![IMG_1489](https://user-images.githubusercontent.com/1490116/176199030-2cfa0b09-2aa9-45aa-9e78-e6134bfe3356.JPG)
![IMG_3094](https://user-images.githubusercontent.com/1490116/176199035-c2d31777-796c-440b-af55-0acf5153db99.JPG)
![IMG_3095](https://user-images.githubusercontent.com/1490116/176199037-37dac8fb-caea-4ba9-8a57-79519786f550.JPG)
![IMG_1499](https://user-images.githubusercontent.com/1490116/178326886-419b0a7e-0ac9-40f8-8916-3847f9ba2464.JPG)
![IMG_3096](https://user-images.githubusercontent.com/1490116/176199040-c377ddbc-d3d9-434b-a1aa-5af02c7b5ba5.JPG)
![IMG_3097](https://user-images.githubusercontent.com/1490116/176199042-40eee728-028c-4765-b71d-5e34ed7e8ebe.JPG)
![IMG_3099](https://user-images.githubusercontent.com/1490116/176199046-c1090fbe-8715-4020-a682-f35d92292526.JPG)

**UPDATE 2 July 2022**
The solar panels specified in the BOM have been out of stock for quite a while, but I finally found panels that were close enough.  UVA CubesatSim #1 (Orange) is finished!  (The +X solar panel looks strange because it's reflecting the blue paper, the tabletop, and the carpet on the floor.)

![CubesatSimWithSolarPanels](https://user-images.githubusercontent.com/1490116/177002046-793d9981-d5d6-4a5c-a070-322b6a884ce8.jpg)

With the CubesatSim on the turntable and the spotlight a few inches away, the voltage curve is really pretty.  (Ignore the first few cycles; the room lights were still on and mostly illuminating the side opposite the spotlight.

![X+Voltage](https://user-images.githubusercontent.com/1490116/177002386-222bc8d0-84e2-4517-a6b0-53689f0fa41a.png)

The panels are even generating a little current.

![X+Current](https://user-images.githubusercontent.com/1490116/177002752-2f0cfe6c-f886-4611-ac57-10c33f6f0408.png)

**Comment by Boovtech** [comment]
Hi Mike,

Your build looks great!  I am in the LONG process of building mine and have had bad issues with the purple boards.  I just bought a whole bunch more and they are costing a lot more than last year.  But a friend already bought a bunch of the blue ones and Alan also suggested using them as well, even though size is an issue, so I bought some and will rebuild a new main board with these.  Now what to do with 15 spare purple ones?!  I guess I will feed the to my herd of hungry Boov!

I felt like I was so close to getting my simulator done until I tested the sensors and had all kinds of issues.  Still, the Pi works great and the process was an education in itself.

Thanks for the closeups on how you got the blue boards to fit.  The thought of starting over is daunting to say the least.  At least the battery board and solar panels are in good shape.  

I see you went with the Pi camera option.  I am tempted to add that to mine since I am having to wait on parts to get going again.

Your aerospace engineering program sounds great!  I'm in the early stages of a BE:EE program with a focus in wireless communication system design.  Aerospace radio is where it's at for me!  I'm a member of AMSAT and love to work satellites when I can.

Thanks again for your notes.  I'll try to be brave enough to post some of my observations here soon.

73
Virginia
NV5F

**Comment by MikeMcPherson** [comment]
Thanks for the very kind comments, Virginia.  I'm glad my writeup was helpful.  I used a Dremel tool to notch the blue INA boards, by the way.  Something is seriously wrong with the purple boards.  I tested 10 of them, and all were dead.  Every blue board worked perfectly the first time.

Incidentally, I had similar trouble with the Blue Pill STM32 boards on the STEM module.  The four I ordered all had defective USB interfaces, which is why I tried the Black Pill.

Best of success with your build and with your BSEE program!

73, Mike KQ9P

**Reply by Boovtech** [reply]
Thanks Mike

It's been slow going here on all fronts for awhile.  I hope things pick up for me this fall.  

I appreciate any tips and the heads-up on bad boards.  These purple things are like $5 a piece and now it looks like I'm kinda stuck with em!  I guess I can test them and try to resell the good ones.  I have another friend who wants to build this too.  But he's far enough behind me that I can run some interference and share all my spare parts!

I'll try to post my progress soon.

73
Virginia

**Comment by MikeMcPherson** [comment]


**Reply by Boovtech** [reply]
HI Mike,

Can you take some photos of your main board from the side so I can see how you mounted your blue boards.  I have finally got all the parts I need to continue my work and I wanted to see how you dealt with this well in advance of trying it myself.

Thanks
Virginia

**Reply by MikeMcPherson** [reply]
Hi Virginia

I added a photo to the original post.  I hope it helps.

Mike

**Reply by Boovtech** [reply]
Thanks Mike.

That photo does help.  It was kind of what I was thinking of doing (except in reverse!).  And from what you said above you just soldered the INA board directly onto the Pi side?  I know it has to be lower because the PI has to clear it.  Also the version of the main board that I have now has the test lead for the sensors, so i I can test them before installing them I will not be so nervous abut having to do this.

Virginia
